gossip-transport-udp/src/main/java/org/apache/gossip/transport/udp/UdpTransportManager.java

```diff
@@ -58,10 +62,35 @@ public UdpTransportManager(GossipManager gossipManager, GossipCore gossipCore) {
       LOGGER.warn(ex);
       throw new RuntimeException(ex);
     }
+    me = new Thread(this);
   }
 
   @Override
+  public void run() {
+    while (keepRunning.get()) {
+      try {
+        byte[] buf = read();
+        try {
+          Base message = gossipManager.getProtocolManager().read(buf);
+          gossipCore.receive(message);
+          //TODO this is suspect
+          gossipManager.getMemberStateRefresher().run();
+        } catch (RuntimeException ex) {//TODO trap json exception
+          LOGGER.error("Unable to process message", ex);
+        }
+      } catch (IOException e) {
+        // InterruptedException are completely normal here because of the blocking lifecycle.
+        if (!(e.getCause() instanceof InterruptedException)) {
+          LOGGER.error(e);
+        }
+        keepRunning.set(false);
+      }
+    }
+  }
+  
+  @Override
   public void shutdown() {
+    keepRunning.set(false);
```

Comment:
If you think that is the right approach. Previously we have been simply letting it fall out of the loop.